src/app/app.component.ts

```diff
@@ -93,3 +131,29 @@ export class HeaderComponent<D> implements OnInit {
   }
 }
 
+const makeCalToken = () => {
+  console.log('making cal');
+  return new BehaviorSubject('Gregorian');
+};
+
+@Directive({
+  selector: 'jdn-datepicker',
+  providers: [
+    { provide: ACTIVE_CALENDAR, useFactory: makeCalToken },
```

Comment:
@mmalerba I think I could implement what you suggested in https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/pull/9#discussion_r316888387.